server/src/main/java/org/elasticsearch/indices/recovery/RecoverySourceHandler.java

```diff
@@ -820,4 +827,30 @@ private void handleErrorOnSendFiles(Store store, Exception e, StoreFileMetaData[
     protected void failEngine(IOException cause) {
         shard.failShard("recovery", cause);
     }
+
+    private <Response> ActionListener<Response> withGenericExecutor(ActionListener<Response> listener) {
```

Comment:
Use ThreadedActionListener instead?

Reply:
Thanks, that was what I was looking for.